src/errors.ts

```diff
@@ -0,0 +1,112 @@
+import type { IncomingHttpHeaders } from "node:http";
+
+export type TimeoutEvent =
+  | "request"
+  | "response"
+  | "read"
+  | "socket"
+  | "lookup"
+  | "connect";
+
+export class StdError extends Error {
+  code?: string;
+  host?: string;
+  hostname?: string;
+  method?: string;
+  path?: string;
+  protocol?: string;
+  url?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- This is a generic error
+  response?: any;
+}
+
+export class RequestError extends StdError {
+  name: "RequestError" = "RequestError";
+}
+
+export class ReadError extends StdError {
+  name: "ReadError" = "ReadError";
+}
+
+export class ParseError extends StdError {
+  name: "ParseError" = "ParseError";
+  statusCode: number;
+  statusMessage: string;
+
+  constructor(
+    message: string,
+    statusCode: number,
+    statusMessage: string
+  ) {
+    super(message);
+    this.statusCode = statusCode;
+    this.statusMessage = statusMessage;
+  }
+}
+
+export class HTTPError extends StdError {
+  name: "HTTPError" = "HTTPError";
+  statusCode: number;
+  statusMessage: string;
+  headers: IncomingHttpHeaders;
+  body?: Buffer | string | object | null;
```

Comment:
Optional AND null? Choose one or the other.

I'd go null personally so it has to be explicit.